javacutil/src/org/checkerframework/javacutil/TypeAnnotationUtils.java

```diff
@@ -65,6 +65,25 @@ public static boolean isTypeCompoundContained(Types types, List<TypeCompound> li
         return false;
     }
 
+    /**
+     * Check whether a TypeCompound with the same name and position is contained in a list of TypeCompounds.
+     *
+     *
+     * @param list The input list of TypeCompounds.
+     * @param tc The TypeCompound to find.
+     * @return true, iff a TypeCompound with the same name and position as tc is contained in list.
+     */
+    public static boolean isTypeCompoundContainedByName(Types types, List<TypeCompound> list, TypeCompound tc) {
+        for (Attribute.TypeCompound rawat : list) {
+            if (contentEquals(rawat.type.tsym.name, tc.type.tsym.name) &&
+                isSameTAPositionExceptTreePos(rawat.position, tc.position)) {
```

Comment:
The tests pass on my viewpoint adaptation branch if I comment out the following line below:
`p1.location.equals(p2.location) &&`